### ISSUE TITLE
install script is not working when api returns rate limit

### ISSUE THREAD

**Issue by Seraphli** [SteamDeckHomebrew/decky-installer#16]
decky loader is not loaded after I update to beta channel (3.6.8). I try to install pre-release version. The installation didn't pop up any error. But the decky loader still didn't show up.
There is a jq error.
`jq: error (at <stdin>:1): Cannot index string with string prerelease`.
It seems accessing `https://api.github.com/repos/SteamDeckHomebrew/decky-loader/releases` returns a rate limit error, so the script didn't work.

**Comment by TrainDoctor** [comment]
@PartyWumpus if you could add some logic to detect the API rate limit that would be appreciated. @Seraphli, we can't do much about the rate limit itself though since that's a GitHub issue and we don't and won't pay for additional API access.

**Comment by Seraphli** [comment]
Yeah I get it. It would be good if this script could add an input for the GitHub token, or you could manually fill in the version. Alternatively, it could just check if the download fails and skip the subsequent operations if it does. This way, it can prevent the deletion of the current loader.

**Comment by TrainDoctor** [comment]
> Yeah I get it. It would be good if this script could add an input for the GitHub token, or you could manually fill in the version. Alternatively, it could just check if the download fails and skip the subsequent operations if it does. This way, it can prevent the deletion of the current loader.

The script does not delete anything until it finishes the download to my knowledge but Party is the maintainer for the installer and would need to confirm that.

**Comment by TeddyKagurazaka** [comment]
Same here,sorry for the bump.
If api.github.com failed because of rate limit, jq -r will fail too, so no download url will get, then nothing will be downloaded.

Check for the download url for empty string should be enough, also check for the existance of PluginLoader after curl downloading.
